p2p/server/msgserver.go

```diff
@@ -162,34 +164,40 @@ func (p *MessageServer) Close() {
 
 // readLoop reads incoming messages and matches them to requests or responses.
 func (p *MessageServer) readLoop(ctx context.Context) error {
+	logger := p.WithContext(ctx)
 	sctx := log.WithNewSessionID(ctx)
 	timer := time.NewTicker(p.requestLifetime + time.Millisecond*100)
 	defer timer.Stop()
-	defer p.With().Info("shutting down protocol", log.String("protocol", p.name))
+	defer logger.With().Info("shutting down protocol", log.String("protocol", p.name))
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-timer.C:
-			p.tg.Go(func(ctx context.Context) error {
+			if err := p.tg.Go(func(ctx context.Context) error {
 				p.cleanStaleMessages()
 				return nil
-			})
+			}); err != nil {
+				logger.With().Panic("error from cleanStaleMessages taskgroup", log.Err(err))
+			}
 		case msg, ok := <-p.ingressChannel:
 			// generate new reqID for message
 			ctx := log.WithNewRequestID(ctx)
-			p.WithContext(ctx).Debug("new msg received from channel")
+			// this causes issues with tests, leaving it here for debugging purposes
+			//logger.Debug("new msg received from channel")
 			if !ok {
-				p.WithContext(ctx).Error("read loop channel was closed")
+				logger.Error("read loop channel was closed")
 				return context.Canceled
 			}
 			select {
 			case p.workerLimiter <- struct{}{}:
-				p.tg.Go(func(ctx context.Context) error {
+				if err := p.tg.Go(func(ctx context.Context) error {
 					p.handleMessage(sctx, msg.(Message)) // pass session ctx to log session id
 					<-p.workerLimiter
 					return nil
-				})
+				}); err != nil {
+					logger.With().Panic("error from handleMessage taskgroup", log.Err(err))
```

Comment:
the same. tbh i don't under5tand why you need to check this error. this is a part of the contract that this error is optional

Reply:
to explain a bit. this error is needed only if you want to know if goroutine was started or not. and for example if it wasn't started you know that tg was stopped and can exit immediately. printing this information is not useful at all

Reply:
Good to know, but I'm still a bit confused. Don't we want to know if the taskgroup was stopped, and therefore the task failed to schedule? I've reduced this from a panic to info, is that what you had in mind?